src/components/services/AccountCallbackManager.tsx

```diff
@@ -86,6 +87,28 @@ export function AccountCallbackManager(props: Props) {
       }),
 
       wallet.on('newTransactions', transactions => {
+        for (const tx of transactions) {
+          const txid = tx.txid.toLowerCase()
+          const cacheEntries = stakeMetadataCache[txid]
+          // Assign cached stake metadata
+          if (cacheEntries != null) {
+            cacheEntries.forEach(cacheEntry => {
+              const { currencyCode, metadata, nativeAmount } = cacheEntry
+
+              const newTx: EdgeTransaction = {
+                ...tx,
+                currencyCode,
+                nativeAmount: nativeAmount ?? tx.nativeAmount,
```

Comment:
Only some places in the cemetery policy is using this nativeAmount override. So this will only impact Tomb. The reason for this is because a single transaction impacts three wallets at once: tokenA, tokenB, and mainnet. For both token wallets, there is not nativeAmount for the tx because it's not an ERC-20 `transfer` call but a LP deposit. I think at least at the time of writing Tomb, the plugin didn't support inferring these amounts from this contract method.